data/mods/ssb/moves.ts

```diff
@@ -369,2444 +593,2591 @@ export const Moves: {[k: string]: ModdedMoveData} = {
+						const newSlot = {
+							id: newMove.id,
+							move: newMove.name,
+							pp: newMove.pp * 8 / 5,
+							maxpp: newMove.pp * 8 / 5,
+							disabled: slot.disabled,
+							used: false,
+						};
+						return newSlot;
+					}
+					return slot;
+				});
+			} else {
+				let transformingLetter = source.species.id[0];
+				if (transformingLetter === 'a') transformingLetter = '';
+				source.formeChange(`unown${transformingLetter}`, this.effect, true);
+				source.moveSlots = source.moveSlots.map(slot => {
+					const newMoves: {[k: string]: string} = {
+						strangesteam: 'lightofruin',
+						psyshock: 'psystrike',
+						aurasphere: 'secretsword',
+						flamethrower: 'mindblown',
+						appleacid: 'seedflare',
+					};
+					if (slot.id in newMoves) {
+						const newMove = this.dex.getMove(newMoves[slot.id]);
+						const newSlot = {
+							id: newMove.id,
+							move: newMove.name,
+							pp: newMove.pp * 8 / 5,
+							maxpp: newMove.pp * 8 / 5,
+							disabled: slot.disabled,
+							used: false,
+						};
+						return newSlot;
+					}
+					return slot;
+				});
+			}
 		},
 		secondary: null,
-		target: "self",
-		type: "Normal",
-
+		target: "normal",
+		type: "Dark",
 	},
-	// LifeisDANK
-	barfight: {
+
+	// Lamp
+	soulswap: {
 		accuracy: 100,
-		basePower: 10,
-		category: "Physical",
-		desc: "Raises both the user's and the target's Attack by three stages, lowers the Defense of both by three stages, confuses both Pokemon, and has a 100% chance to cause the target to flinch. Only works on the user's first turn on the field.",
-		shortDesc: "First turn: +3 Atk, -3 Def, flinch, both confused.",
-		name: "Bar Fight",
+		basePower: 90,
+		category: "Special",
+		desc: "The user copies the target's positive stat stage changes and then inverts the target's stats.",
+		shortDesc: "Copies target's stat boosts then inverts.",
+		name: "Soul Swap",
 		isNonstandard: "Custom",
+		gen: 8,
 		pp: 10,
-		priority: 3,
-		flags: {protect: 1, mirror: 1, contact: 1},
-		onTry(pokemon, target) {
-			if (pokemon.activeMoveActions > 1) {
-				this.attrLastMove('[still]');
-				this.add('-fail', pokemon);
-				this.hint("Bar Fight only works on your first turn out.");
-				return null;
-			}
-		},
+		priority: 0,
+		flags: {protect: 1, mirror: 1},
 		onTryMove() {
 			this.attrLastMove('[still]');
 		},
 		onPrepareHit(target, source) {
-			this.add('-anim', source, "Fake Out", target);
-			this.add('-anim', source, "Feather Dance", target);
+			this.add('-anim', source, 'Spectral Thief', target);
+			this.add('-anim', source, 'Teleport', source);
+			this.add('-anim', source, 'Topsy-Turvy', target);
 		},
 		onHit(target, source) {
-			this.boost({atk: 3, def: -3}, target);
-			this.boost({atk: 3, def: -3}, source);
-			target.addVolatile('confusion');
-			source.addVolatile('confusion');
-			target.addVolatile('flinch');
+			let i: BoostName;
+			const boosts: SparseBoostsTable = {};
+			for (i in target.boosts) {
```

Comment:
The `let i: BoostName;` should always be the line before the `for (i in target.boosts) {`. This will make it easier to fix if TypeScript changes things later.